server/application/application.go

```diff
@@ -987,6 +993,23 @@ func (s *Server) updateApp(ctx context.Context, app *v1alpha1.Application, newAp
 	return nil, status.Errorf(codes.Internal, "Failed to update application. Too many conflicts")
 }
 
+func diffBetweenApplicationSpecs(a *v1alpha1.ApplicationSpec, b *v1alpha1.ApplicationSpec) (string, error) {
+	aBytes, err := json.Marshal(a)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal application: %w", err)
+	}
+	bBytes, err := json.Marshal(b)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal application: %w", err)
+	}
+
+	patch, err := jsonpatch.CreateMergePatch(aBytes, bBytes)
+	if err != nil {
+		return "", fmt.Errorf("failed to create json diff between applications: %w", err)
+	}
+	return string(patch), nil
```

Comment:
Would it be possible to unmarshal the patch into a map[string]any so log tools aren't forced to parse a JSON string?